### PR TITLE
things

### PR DESCRIPTION
**Pull request by zdimension**
added BadFloor, ClearSpiders, LoadRope, SaveRope, StraightenRope, GetFreeBlood, Richochet

### CODE REVIEW

File: GAME/BOX.C [hunk 1]
```diff
@@ -434,10 +434,36 @@ void CreatureDie(short item_number, int explode)// (F)
 	}
 }
 
-int BadFloor(long x, long y, long z, long box_height, long next_height, int room_number, struct lot_info* LOT)
+int BadFloor(long x, long y, long z, long box_height, long next_height, int room_number, struct lot_info* LOT)// (F)
 {
-	S_Warn("[BadFloor] - Unimplemented!\n");
-	return 0;
+	short room_num = room_number;
```
Comment (Gh0stBlade) [inline GAME/BOX.C]: Variable not needed, psx just takes room_number. No additional var.

Comment (zdimension) [reply GAME/BOX.C]: Variable is needed because room_number is an int and GetFloor expects a short*

Comment (Gh0stBlade) [reply GAME/BOX.C]: Why not cast? I'd assume that's what they did originally?

floor = GetFloor(x, y, z, (short*) &room_number);

Comment (zdimension) [reply GAME/BOX.C]: That code would break on a big-endian system

It just happens to work because fortunately PSX and PC use little endian

Comment (Gh0stBlade) [reply GAME/BOX.C]: Approved.

**Pull request merged**